Python 3.6.5 |Anaconda custom (64-bit)| (default, Mar 29 2018, 13:32:41) [MSC v.1900 64 bit (AMD64)] on win32
Type "copyright", "credits" or "license()" for more information.
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
>>> print(array)
[[1 2 3]
 [2 3 4]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [0 0 0]]
number of dim:  2
shape:  (2, 3)
size:  6
>>> type(array)
<class 'numpy.ndarray'>
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2 23  4]
>>> type(a)
<class 'numpy.ndarray'>
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0. 0. 0. 0.]
 [0. 0. 0. 0.]
 [0. 0. 0. 0.]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0 0 0 0]
 [0 0 0 0]
 [0 0 0 0]] int16
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0 0 0 0]
 [0 0 0 0]
 [0 0 0 0]] int16
[10 12 14 16 18]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0 0 0 0]
 [0 0 0 0]
 [0 0 0 0]] int16
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1210, in <module>
    c=np.arange(10,20,2).reshape((3,4))
ValueError: cannot reshape array of size 5 into shape (3,4)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0 0 0 0]
 [0 0 0 0]
 [0 0 0 0]] int16
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1210, in <module>
    c=np.arange(10,20,1).reshape((3,4))
ValueError: cannot reshape array of size 10 into shape (3,4)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0 0 0 0]
 [0 0 0 0]
 [0 0 0 0]] int16
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1210, in <module>
    c=np.arange(10,22,2).reshape((3,4))
ValueError: cannot reshape array of size 6 into shape (3,4)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0 0 0 0]
 [0 0 0 0]
 [0 0 0 0]] int16
[[10 12 14]
 [16 18 20]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 2 3]
 [2 3 4]]
number of dim:  2
shape:  (2, 3)
size:  6
[ 2. 23.  4.]
float64
[[0 0 0 0]
 [0 0 0 0]
 [0 0 0 0]] int16
[[10 12 14]
 [16 18 20]]
[[ 1.          1.81818182  2.63636364  3.45454545]
 [ 4.27272727  5.09090909  5.90909091  6.72727273]
 [ 7.54545455  8.36363636  9.18181818 10.        ]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[-5.44021111  9.12945251 -9.88031624  7.4511316 ]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[-5.44021111  9.12945251 -9.88031624  7.4511316 ]
[0 1 2 3]
[ True  True  True False]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[-5.44021111  9.12945251 -9.88031624  7.4511316 ]
[0 1 2 3]
[ True  True  True False]
[False False False  True]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0 1]
 [0 3]]
[[2 4]
 [2 3]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0 1]
 [0 3]]
[[2 4]
 [2 3]]
[[2 4]
 [2 3]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
>>> print(a)
0.40894631110627455
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[0.64039659]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[0.84494719 0.51059277]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[0.2404765 0.5606076]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0.16279131 0.21635406 0.07076724]
 [0.44148004 0.1440137  0.45407653]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0.90694631 0.22150067 0.11007229 0.1923891 ]
 [0.2539481  0.2647268  0.63240291 0.26237373]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0.83848943 0.36747996 0.36133139 0.83840678]
 [0.06308398 0.37040384 0.08953924 0.09649282]]
3.025227432951467
0.06308398204788968
0.8384894250314888
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0.72690962 0.3534445  0.12411078 0.02681142]
 [0.77437142 0.01958289 0.55001218 0.59871812]]
[1.23127633 1.94268461]
[0.72690962 0.01958289 0.12411078 0.02681142]
[0.72690962 0.77437142]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0.49144261 0.08220291 0.65506486 0.55473259]
 [0.41344043 0.63753252 0.44442496 0.62821248]]
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1256, in <module>
    print(np.sum(a,axis=2))     #-----------axis=1为行；=0为列
  File "F:\anaconda\lib\site-packages\numpy\core\fromnumeric.py", line 1882, in sum
    out=out, **kwargs)
  File "F:\anaconda\lib\site-packages\numpy\core\_methods.py", line 32, in _sum
    return umr_sum(a, axis, dtype, out, keepdims)
numpy.core._internal.AxisError: axis 2 is out of bounds for array of dimension 2
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[0.08459557 0.07339529 0.60401961 0.47607317]
 [0.59295864 0.28127113 0.06411438 0.86882934]]
[1.23808364 1.80717349]
[0.08459557 0.07339529 0.06411438 0.47607317]
[0.60401961 0.86882934]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1275, in <module>
    print(A.median())
AttributeError: 'numpy.ndarray' object has no attribute 'median'
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
7.5
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
7.5
[ 2  5  9 14 20 27 35 44 54 65 77 90]
[[1 1 1]
 [1 1 1]
 [1 1 1]]
(array([0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2], dtype=int64), array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3], dtype=int64))
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
7.5
[ 2  5  9 14 20 27 35 44 54 65 77 90]
[[1 1 1]
 [1 1 1]
 [1 1 1]]
(array([0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2], dtype=int64), array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3], dtype=int64))
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]

Warning (from warnings module):
  File "C:\Users\Administrator\Desktop\code.py", line 1288
    print(np.rank(A))               #-----矩阵的秩
VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.
2
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
7.5
[ 2  5  9 14 20 27 35 44 54 65 77 90]
[[1 1 1]
 [1 1 1]
 [1 1 1]]
(array([0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2], dtype=int64), array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3], dtype=int64))
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
2
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
7.5
[ 2  5  9 14 20 27 35 44 54 65 77 90]
[[1 1 1]
 [1 1 1]
 [1 1 1]]
(array([0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2], dtype=int64), array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3], dtype=int64))
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
2
[[5 5 5 5]
 [6 7 8 9]
 [9 9 9 9]]
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1292, in <module>
    print(np.mean(A,aixs=0))               #-----都可指定行 ，列计算
TypeError: mean() got an unexpected keyword argument 'aixs'
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
7.5
[ 2  5  9 14 20 27 35 44 54 65 77 90]
[[1 1 1]
 [1 1 1]
 [1 1 1]]
(array([0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2], dtype=int64), array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3], dtype=int64))
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
2
[[5 5 5 5]
 [6 7 8 9]
 [9 9 9 9]]
[6. 7. 8. 9.]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
0
11
7.5
7.5
7.5
7.5
[ 2  5  9 14 20 27 35 44 54 65 77 90]
[[1 1 1]
 [1 1 1]
 [1 1 1]]
(array([0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2], dtype=int64), array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3], dtype=int64))
[[ 2  3  4  5]
 [ 6  7  8  9]
 [10 11 12 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
[[ 2  6 10]
 [ 3  7 11]
 [ 4  8 12]
 [ 5  9 13]]
2
[[5 5 5 5]
 [6 7 8 9]
 [9 9 9 9]]
[ 3.5  7.5 11.5]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
8
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
8
8
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
3
4
5
6
7
8
9
10
11
12
13
14
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
3
4
5
6
7
8
9
10
11
12
13
14
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
[3 4 5 6]
[ 7  8  9 10]
[11 12 13 14]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
[3 4 5 6]
[ 7  8  9 10]
[11 12 13 14]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
[3 4 5 6]
[ 7  8  9 10]
[11 12 13 14]
[ 3  7 11]
[ 4  8 12]
[ 5  9 13]
[ 6 10 14]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
[3 4 5 6]
[ 7  8  9 10]
[11 12 13 14]
[ 3  7 11]
[ 4  8 12]
[ 5  9 13]
[ 6 10 14]
3
4
5
6
7
8
9
10
11
12
13
14
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[ 3  4  5  6  7  8  9 10 11 12 13 14]
6
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[11 12 13 14]
[11 12 13 14]
8
8
[8 9]
[3 4 5 6]
[ 7  8  9 10]
[11 12 13 14]
[ 3  7 11]
[ 4  8 12]
[ 5  9 13]
[ 6 10 14]
[[ 3  4  5  6]
 [ 7  8  9 10]
 [11 12 13 14]]
[ 3  4  5  6  7  8  9 10 11 12 13 14]
3
4
5
6
7
8
9
10
11
12
13
14
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
(3,) (2, 3)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1334, in <module>
    D=bp.hstack((A,B))
NameError: name 'bp' is not defined
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
(6,)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
[[1 2]
 [1 2]
 [1 2]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
[[1 2]
 [1 2]
 [1 2]]
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1368, in <module>
    C_=np.concatenate((A,B,B,A),axis=1)
numpy.core._internal.AxisError: axis 1 is out of bounds for array of dimension 1
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
[[1 2]
 [1 2]
 [1 2]]
[1 1 1 2 2 2 2 2 2 1 1 1]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
[[1 2]
 [1 2]
 [1 2]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
[[1 2]
 [1 2]
 [1 2]]
[[1 2 2 1]
 [1 2 2 1]
 [1 2 2 1]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
[[1 2]
 [1 2]
 [1 2]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[1 1 1]
 [2 2 2]]
(3,) (2, 3)
[1 1 1 2 2 2]
(6,)
[1 1 1 2 2 2]
(6,)
(1, 3)
[[1 1 1]]
(3, 1)
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]]
[[1]
 [1]
 [1]
 [2]
 [2]
 [2]]
[[1 2]
 [1 2]
 [1 2]]
[[1 2 2 1]
 [1 2 2 1]
 [1 2 2 1]]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
Traceback (most recent call last):
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 553, in split
    len(indices_or_sections)
TypeError: object of type 'int' has no len()

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1381, in <module>
    print(np.split(A,2,axis=0))                  #-----分割
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 559, in split
    'array split does not result in an equal division')
ValueError: array split does not result in an equal division
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1, 2, 3]]), array([[4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
[array([[0, 1, 2, 3],
       [4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
[array([[0, 1, 2, 3],
       [4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
[array([[0, 1, 2, 3]]), array([[4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
[array([[0, 1, 2, 3],
       [4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
[array([[0, 1, 2, 3]]), array([[4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
Traceback (most recent call last):
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 553, in split
    len(indices_or_sections)
TypeError: object of type 'int' has no len()

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1386, in <module>
    print(np.hsplit(A,3))
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 621, in hsplit
    return split(ary, indices_or_sections, 1)
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 559, in split
    'array split does not result in an equal division')
ValueError: array split does not result in an equal division
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
[array([[0, 1, 2, 3]]), array([[4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
[array([[0, 1, 2, 3],
       [4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
Traceback (most recent call last):
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 553, in split
    len(indices_or_sections)
TypeError: object of type 'int' has no len()

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\code.py", line 1382, in <module>
    print(np.vsplit(A,2))
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 674, in vsplit
    return split(ary, indices_or_sections, 0)
  File "F:\anaconda\lib\site-packages\numpy\lib\shape_base.py", line 559, in split
    'array split does not result in an equal division')
ValueError: array split does not result in an equal division
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
[array([[0, 1, 2, 3]]), array([[4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
[array([[0, 1],
       [4, 5],
       [8, 9]]), array([[ 2,  3],
       [ 6,  7],
       [10, 11]])]
[array([[0, 1, 2, 3],
       [4, 5, 6, 7]]), array([[ 8,  9, 10, 11]])]
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
>>> a
array([0, 1, 2, 3])
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
>>> a
array([11,  1,  2,  3])
>>> b
array([11,  1,  2,  3])
>>> c
array([11,  1,  2,  3])
>>> d
array([11,  1,  2,  3])
>>> a==b
array([ True,  True,  True,  True])
>>> c==a
array([ True,  True,  True,  True])
>>> d==a
array([ True,  True,  True,  True])
>>> b is a
True
>>> d is a
True
>>> a is b
True
>>> a is c
True
>>> 
============== RESTART: C:\Users\Administrator\Desktop\code.py ==============
>>> b_
array([11,  1,  2,  3])
>>> a[0]=0
>>> a
array([0, 1, 2, 3])
>>> b
array([0, 1, 2, 3])
>>> c
array([0, 1, 2, 3])
>>> d
array([0, 1, 2, 3])
>>> b_
array([11,  1,  2,  3])
>>> b_ is a
False
>>> type a
SyntaxError: invalid syntax
>>> type a
SyntaxError: invalid syntax
>>> type(a)
<class 'numpy.ndarray'>
>>> type(b_)
<class 'numpy.ndarray'>
>>> 
